### PR TITLE
Songs in posts

### PR DESCRIPTION
**Pull request by camrcast**
Added songs to post and a filter if the artist and/or song title do not match any result from the api

### CODE REVIEW

File: __tests__/postService.test.js [hunk 1]
```diff
@@ -22,7 +24,7 @@ const mockPost2 = {
     postedBy: "6d737a3b-d543-459b-aca6-d1f04952bf30",
     description: "This is a great song",
     score: 100,
-    title: "Title",
+    song: "Title",
```
Comment (hubmelco) [inline __tests__/postService.test.js:27]: Change one of the "song"s back to "title"

File: src/controller/postRouter.js [hunk 2]
```diff
@@ -165,14 +156,6 @@ postRouter.patch("/:postId/likes", authenticate, postMiddleware.validateLike, as
...
 
-/**
- * Deletes a post by their id
- * Path Parameter
- *      :postId {string}
- * Response
- *      200 - Successfully deleted the post by their id
- *      400 - Post with id ${postId} not found
- */
 postRouter.delete("/:postId", postOwnerOrAdminAuthenticate, async (req, res) => {
```
Comment (hubmelco) [inline src/controller/postRouter.js:159]: Route comments should stay. Kyle worked hard on them

File: src/repository/songDAO.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-
+const dotenv = require('dotenv');
+dotenv.config();
```
Comment (hubmelco) [inline src/repository/songDAO.js:2]: This should not be in here. I have it in index.js before anything gets set up so all routes should already have access to environment variables

File: src/services/postService.js [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
-const createPost = async (userId, description, score, title, tags) => {
+const createPost = async (userId, description, score, title, artist, tags) => {
     let tagMap = new Map();
     if (tags){
         for (const i of tags){
             tagMap.set(i, true);
         }
     }
-    const post = { class: CLASS_POST, itemID: uuid.v4(), postedBy: userId, description, score, title, replies: [], likedBy: [], tags: tagMap, isFlagged: 0 };
+    const result = await getSongs({track: title, artist}, 0);
```
Comment (hubmelco) [inline src/services/postService.js:13]: I would make this song request stuff all frontend work. Allow the user to search for a song. They select one, then you take the title and artist and append it to the post body. It's be 2 different backend calls. One for searching songs which are then displayed to the user so they can select one. Then once it is selected, take the data and make a call to create a post with that data. That way there's algorithm to make a best guess at what the user wants. We just simply pass the data in to create a post

**Review by hubmelco — COMMENTED** [review]
great work. I like these small PRs

File: src/controller/postRouter.js (hunk 2 above)
Comment (camrcast) [reply src/controller/postRouter.js:159]: Huh, I do not know  why they got deleted. Yes I will return them.

File: src/controller/postRouter.js [hunk 5]
```diff
@@ -19,10 +19,10 @@ const postRouter = express.Router();
 postRouter.post("/", authenticate, postMiddleware.validateTextBody, postMiddleware.validateScore, async (req, res) => {
     //TODO check song title exists in API
     const userId = res.locals.user.itemID;
-    const { text, score, title, tags } = req.body;
+    const { text, score, song, artist, tags } = req.body;
```
Comment (hubmelco) [inline src/controller/postRouter.js:22]: We still need title. Don't need artist. Its included in the song object

File: src/controller/postRouter.js [hunk 6]
```diff
@@ -19,10 +19,10 @@ const postRouter = express.Router();
...
 
     try {
-        const post = await postService.createPost(userId, text, score, title, tags);
+        const post = await postService.createPost(userId, text, score, song, artist, tags);
```
Comment (hubmelco) [inline src/controller/postRouter.js:25]: Need title, don't need artist

File: src/controller/postRouter.js [hunk 7]
```diff
@@ -58,14 +58,6 @@ postRouter.patch("/:postId", authenticate, async (req, res) => {
...
 
-/**
- * Gets a post by their id
- * Path Parameter
- *      :postId {string}
- * Response
- *      200 - Successfully received the post by their id
- *      400 - Post with id ${postId} not found
- */
 postRouter.get("/:postId", async (req, res) => {
```
Comment (hubmelco) [inline src/controller/postRouter.js:61]: Add comment back

File: src/services/postService.js [hunk 8]
```diff
@@ -2,14 +2,14 @@ const uuid = require("uuid");
 const { throwIfError, CLASS_POST } = require('../utilities/dynamoUtilities');
 const postDAO = require("../repository/postDAO");
 
-const createPost = async (userId, description, score, title, tags) => {
+const createPost = async (userId, description, score, song, tags) => {
```
Comment (hubmelco) [inline src/services/postService.js:5]: Need title

File: src/services/postService.js [hunk 9]
```diff
@@ -2,14 +2,14 @@ const uuid = require("uuid");
...
     let tagMap = new Map();
     if (tags){
         for (const i of tags){
             tagMap.set(i, true);
         }
     }
-    const post = { class: CLASS_POST, itemID: uuid.v4(), postedBy: userId, description, score, title, replies: [], likedBy: [], tags: tagMap, isFlagged: 0 };
+    const post = { class: CLASS_POST, itemID: uuid.v4(), postedBy: userId, description, score, song, replies: [], likedBy: [], tags: tagMap, isFlagged: 0 };
```
Comment (hubmelco) [inline src/services/postService.js:12]: need title

**Review by kylefwalters — APPROVED** [review]

**Pull request merged**